modules/vaos/app/services/vaos/v2/appointments_reason_code_service.rb

```diff
@@ -39,10 +35,18 @@ def extract_reason_code_fields(appointment)
                                            .to_h { |pair| pair.split(':').map!(&:strip) }
         return if reason_code_hash.empty?
 
-        contact = extract_contact_fields(reason_code_hash)
-        comments = reason_code_hash['comments'] if reason_code_hash.key?('comments')
-        reason = extract_reason_for_appointment(reason_code_hash)
-        preferred_dates = extract_preferred_dates(reason_code_hash)
+        # Direct Scheduling appointments
+        if appointment[:kind] == 'clinic' && appointment[:status] == 'booked'
+          comments = reason_code_hash['comments'] if reason_code_hash.key?('comments')
```

Comment:
Coming from a background with stricter language rules, Ruby interpreter scares me a little bit. From my current understanding:
1. local variables's scope is the entire function so variables declared and defined in an if statement is visible outside of it
2. undeclared or undefined variables (e.g. contact for directs scheduling appointments) are assumed to have the value nil

For these reasons, the modifications to `appointment` on lines 51-54 works as expected but seems super strange to me. @cferris32 can you double check my understanding when you're back next week?

Reply:
Your understanding is correct. Local variables will have a scope of the function or class they are declared in. Undeclared/undefined values will default to nil. Similar to the lines you referenced, you should always be able to perform a `.nil?` check on a variable/object/field in order to verify.

If you're ever interested, running `rails console` in your IDE terminal provides a sandbox to test out these theories and develop new layers of understanding. I've found it extremely helpful for things like this.